src/Services/CareTeamService.php

```diff
@@ -0,0 +1,128 @@
+                LEFT JOIN facility ON facility.id=patient.care_team_facility) as careTeam
+                LEFT JOIN uuid_mapping ON uuid_mapping.target_uuid=careTeam.puuid AND uuid_mapping.resource='CareTeam'";
+
+        if (!empty($search)) {
+            $sql .= ' WHERE ';
+            $whereClauses = array();
+            foreach ($search as $fieldName => $fieldValue) {
+                array_push($whereClauses, $fieldName . ' = ?');
+                array_push($sqlBindArray, $fieldValue);
+            }
+            $sqlCondition = ($isAndCondition == true) ? 'AND' : 'OR';
+            $sql .= implode(' ' . $sqlCondition . ' ', $whereClauses);
+        }
+
+        $statementResults = sqlStatement($sql, $sqlBindArray);
+
+        $processingResult = new ProcessingResult();
+        while ($row = sqlFetchArray($statementResults)) {
+            $row['uuid'] = UuidRegistry::uuidToString($row['uuid']);
+            $row['puuid'] = UuidRegistry::uuidToString($row['puuid']);
+            $row['pruuid'] = $row['pruuid'] ? UuidRegistry::uuidToString($row['pruuid']) : null;
+            $row['ouuid'] = $row['ouuid'] ? UuidRegistry::uuidToString($row['ouuid']) : null;
+            $processingResult->addData($row);
+        }
+
+        return $processingResult;
+    }
+
+    /**
+     * Returns a single careTeam record by id.
+     * @param $uuid - The careTeam uuid identifier in string format.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function getOne($uuid)
+    {
+        $processingResult = new ProcessingResult();
+
+        $isValid = BaseValidator::validateId(
+            "uuid",
+            "uuid_mapping",
+            $uuid,
+            true
+        );
+
+        $sql = "SELECT careTeam.*, uuid_mapping.uuid as uuid FROM 
+                (SELECT patient.uuid as puuid,
+                users.fname as prac_name,
+                users.uuid as pruuid,
+                facility.name as fac_name,
+                facility.uuid as ouuid
+                FROM patient_data as patient
+                LEFT JOIN users ON users.id=patient.care_team_provider
+                LEFT JOIN facility ON facility.id=patient.care_team_facility) as careTeam
+                LEFT JOIN uuid_mapping ON uuid_mapping.target_uuid=careTeam.puuid AND uuid_mapping.resource='CareTeam'
+                WHERE uuid_mapping.uuid = ?";
+
+        $uuidBinary = UuidRegistry::uuidToBytes($uuid);
+        $sqlResult = sqlQuery($sql, [$uuidBinary]);
+        $sqlResult['uuid'] = UuidRegistry::uuidToString($sqlResult['uuid']);
+        $sqlResult['puuid'] = UuidRegistry::uuidToString($sqlResult['puuid']);
+        $sqlResult['pruuid'] = UuidRegistry::uuidToString($sqlResult['pruuid']);
+        $sqlResult['ouuid'] = UuidRegistry::uuidToString($sqlResult['ouuid']);
+        $processingResult->addData($sqlResult);
+        return $processingResult;
+    }
+}
```

Comment:
will the logic here be more complicated now that there are multiple providers or facilities stored in a field?

Reply:
Yea, I didn't test APIs after the latest change so couldn't be caught this one. Will update the logic tomorrow :+1: 

Thanks :)